core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -239,6 +239,10 @@ abstract class RawModule extends BaseModule {
       case (false, false)                               =>
         // For non-probe, directly create Nodes for lhs, skipping visibility check to support BoringUtils.drive.
         (left, right) match {
+          case (lhsOpaque: OpaqueType, rhsOpaque: OpaqueType)
+              if DataMirror
+                .isProperty(lhsOpaque.allElements.head) && DataMirror.isProperty(rhsOpaque.allElements.head) =>
```

Comment:
I'm not sure if there is a better way to query the underlying Data than `allElements.head`... if this is about the best we can do, maybe I can add an internal-only helper method to access this?

Reply:
See [my comment](https://github.com/chipsalliance/chisel/pull/4337/files#r1712166891) on all 4 lines with suggestion.